fix(components): [table] height为百分比的时候，inner-wrapper高度异常

WindBearr opened the pull request:
当el-table的height="50%",那么不仅el-table高度变为了50%,其内部div高度也被赋予了50%，这样会导致样式需要手动去调节css兼容才符合直觉。

所以针对height的复制，存在"%"时，将内部div高度调整为'100%'，填充满整个table。兼容height的百分比高度。

BREAKING CHANGE :
没有N

closed 有Y:  #9536

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

WindBearr committed:
fix(components): 增加了一个输出选项，让表格百分比高度得到正确显示

当el-table的height="50%",那么不仅el-table高度变为了50%,其内部div高度也被赋予了50%，这样会导致样式需要手动去调节css兼容才符合直觉。

所以针对height的复制，存在"%"时，将内部div高度调整为'100%'，填充满整个table。兼容height的百分比高度。

BREAKING CHANGE :
没有N

closed 有Y:  #9536

WindBearr commented:
#9536

jw-foss commented on packages/components/table/src/table/style-helper.ts:
```diff
@@ -284,18 +284,20 @@ function useStyle<T>(
+    let height = props.height || props.maxHeight
+    if (!Number.isNaN(Number(props.height))) {
+      height = `${height}px`
+    } else if (props.height.includes('%')) {
```
Could you please elaborate on why when the `height` is in percentage we set it to `100%`? What was the point of passing a percentage value to this component if we do that?

WindBearr replied:
1.What was the point of passing a percentage value to this component if we do that? 
example:   <el-table height="50%"> 是为了让<div class="el-table" height="50%">， 而不是让<div class="el-table__inner-wrapper">的高度也设置为50%。 这个样式不应该继续传递给内部<div>, 会让表格内部出现空白的问题。这个问题非常容易复现。在实际开发中，我必须设置height="100%"，不然就会出现空白问题。

2.Could you please elaborate on why when the height is in percentage we set it to 100%？
Answer: 我认为更好的做法是不去设置<div class="el-table__inner-wrapper">的行内样式。 我们并不需要 'tableInnerStyle'。 因为有 css 去设置 class="el-table__inner-wrapper" 的样式。

3. Why would I modify the code like this?
因为我不知道直接去除‘tableInnerStyle’会不会造成其他bug， 所以将'tableInnerStyle'返回的所有百分比设置100%。
目前的解决方法不会给其他代码造成影响，而且可以解决给组件设置百分比高度造成的问题。

WindBearr replied:
如果你同意第二种更好的方法，我可以采用第二种方法，然后重新提交一次代码合并请求

WindBearr committed:
Merge branch 'dev' into dev

WindBearr committed:
Merge branch 'element-plus:dev' into dev

WindBearr committed:
Merge branch 'dev' into dev

WindBearr committed:
Merge branch 'element-plus:dev' into dev

WindBearr committed:
style(theme-chalk): added a line of code

When you hover over the lower part of a menu-item, the next menu-item will be selected

Because the height of menu-item is 36px, but the line-height is 56px.

Set the line-height to the same height as the height

closed N
